llama-index-integrations/embeddings/llama-index-embeddings-ipex-llm/llama_index/embeddings/ipex_llm/base.py

```diff
@@ -57,7 +57,15 @@ def __init__(
         callback_manager: Optional[CallbackManager] = None,
         **model_kwargs,
     ):
-        self._device = device or infer_torch_device()
+        # Set "cpu" as default device
+        self._device = device or "cpu"
+
+        if self._device not in ["cpu", "xpu"]:
+            logger.warning(
+                "IpexLLMEmbedding currently only supports device to be 'cpu' or 'xpu', "
+                f"but you have: {self._device}; Use 'cpu' instead."
+            )
+            self._device = "cpu"
```

Comment:
In the above we change the parameter type of `device` in the __init__ constructor, and rather than logging a warning, we raise an error if an invalid device type is supplied. If no `device` is supplied at construction, then `cpu` is the default value.

Reply:
Thanks a lot for the comments! Have updared :)